test/integration/addons_test.go

```diff
@@ -598,10 +598,25 @@ func validateCSIDriverAndSnapshots(ctx context.Context, t *testing.T, profile st
 	}
 }
 
+// validateGCPAuthNamespaces validates that newly created namespaces contain the gcp-auth secret.
+func validateGCPAuthNamespaces(ctx context.Context, t *testing.T, profile string) {
+	rr, err := Run(t, exec.CommandContext(ctx, "kubectl", "--context", profile, "create", "ns", "new-namespace"))
+	if err != nil {
+		t.Fatalf("%s failed: %v", rr.Command(), err)
+	}
+
+	rr, err = Run(t, exec.CommandContext(ctx, "kubectl", "--context", profile, "get", "secret", "gcp-auth", "-n", "new-namespace"))
+	if err != nil {
+		t.Errorf("%s failed: %v", rr.Command(), err)
+	}
+}
+
 // validateGCPAuthAddon tests the GCP Auth addon with either phony or real credentials and makes sure the files are mounted into pods correctly
 func validateGCPAuthAddon(ctx context.Context, t *testing.T, profile string) {
 	defer PostMortemLogs(t, profile)
 
+	validateGCPAuthNamespaces(ctx, t, profile)
```

Comment:
This func should be wrapped in:
```
t.Run("AuthNamespace", func(t *testing.T) {
   validateGCPAuthNamespaces(ctx, t, profile)
})
```
As currently the failure will be reported in the parent test so the `t.Fatalf` will kill the entire GCP-Auth test instead of just the namespace test itself.

Reply:
hmm I just figured I'd pull this code into a separate function since the gcpAuth test is already quite long. I could create a subtest as you suggest or I could just inline it to be consistent. What do you think?